### PR TITLE
Refactor AccelerometerThread.h 

### PR DESCRIPTION
**Pull request by dhskinner**
This PR is to refactor AccelerometerThread.h into its own folder and tease apart the code into individual classes per sensor to make it easier to add more sensors in future - for more info refer to #4662 

It picks up all 6 existing accelerometers, plus adds a basic wake-on-move implementation for a new sensor ICM-20948 (the intent is add digital motion processing for ICM-20948 in a new PR to come)

### CODE REVIEW

File: src/motion/AccelerometerThread.h [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+  public:
+    explicit AccelerometerThread(ScanI2C::FoundDevice found) : OSThread("AccelerometerThread")
+    {
+        if (found.address.port == ScanI2C::I2CPort::NO_I2C) {
+            LOG_DEBUG("AccelerometerThread disabling due to no sensors found\n");
+            disable();
+            return;
+        }
+
+#ifndef RAK_4631
```
Comment (fifieldt) [inline src/motion/AccelerometerThread.h:35]: Q) Does RAK have magic accelerometers that are always in an interesting configuration? :)

File: src/motion/STK8XXXSensor.h [hunk 2]
```diff
@@ -0,0 +1,37 @@
+#pragma once
+#ifndef _STK8XXX_SENSOR_H_
+#define _STK8XXX_SENSOR_H_
+
```
Comment (fifieldt) [inline src/motion/STK8XXXSensor.h:4]: Should there be an #endif here, rather than at the end of the file?

**Comment by fifieldt** [comment]
This is a significant improvement. Many many thanks.

File: src/detect/ScanI2CTwoWire.cpp [hunk 3]
```diff
@@ -397,6 +394,24 @@ void ScanI2CTwoWire::scanPort(I2CPort port, uint8_t *address, uint8_t asize)
...
+                registerValue = getRegisterValue(ScanI2CTwoWire::RegisterLocation(addr, 0x00), 1);
+                if (registerValue == 0xEA) {
+                    type = ICM20948;
+                    LOG_INFO("ICM20948 9-dof motion processor found\n");
+                    break;
+                } else if (addr.address == BMX160_ADDR) {
+                    type = BMX160;
+                    LOG_INFO("BMX160 accelerometer found\n");
+                    break;
+                } else if (addr.address == MPU6050_ADDR) {
```
Comment (caveman99) [inline src/detect/ScanI2CTwoWire.cpp:408]: logically this is the last possible condition, that can also be a simple 'else'

File: src/main.cpp [hunk 4]
```diff
@@ -638,7 +639,7 @@ void setup()
 #if !MESHTASTIC_EXCLUDE_I2C
 #if !defined(ARCH_PORTDUINO) && !defined(ARCH_STM32WL) && !MESHTASTIC_EXCLUDE_ENVIRONMENTAL_SENSOR
     if (acc_info.type != ScanI2C::DeviceType::NONE) {
-        accelerometerThread = new AccelerometerThread(acc_info.type);
+        accelerometerThread = new AccelerometerThread(acc_info);
     }
```
Comment (caveman99) [inline src/main.cpp:643]: Didn't check - did you change the interface?

File: src/motion/AccelerometerThread.h (hunk 1 above)
Comment (caveman99) [reply src/motion/AccelerometerThread.h:35]: 4631 always uses tap and double tap functionaltiy when the module is detected

File: src/modules/AdminModule.cpp [hunk 5]
```diff
@@ -34,10 +31,6 @@
 #include "modules/PositionModule.h"
 #endif
 
-#if !defined(ARCH_PORTDUINO) && !defined(ARCH_STM32WL) && !MESHTASTIC_EXCLUDE_ENVIRONMENTAL_SENSOR
-#include "AccelerometerThread.h"
-#endif
-
```
Comment (caveman99) [inline src/modules/AdminModule.cpp:40]: Is this just some leftover cleanup like the main.h above or was this required for actual functionality?

**Review by caveman99 — CHANGES_REQUESTED** [review]

**Comment by caveman99** [comment]
Just a general note - Please enable maintainer edits on your PR/Branch, in the current state it is impossible to merge this.

**Comment by dhskinner** [comment]
> Just a general note - Please enable maintainer edits on your PR/Branch, in the current state it is impossible to merge this.

@caveman99 apologies I'm going to need to resubmit this as a new PR.... super frustrating as it turns out I can't "enable edits from maintainers" from an organisation account because of [this issue here](https://github.com/orgs/community/discussions/5634)

Big sigh / didn't know this limitation was a thing

**Comment by caveman99** [comment]
god, i didn't know that was a problem, sorry. The maintainer edit rights are not used to alter the PR in any form (save the odd trunk fmt) but we need it cause we have a 'aonly merge if the PR is rebased to top of master' lock i can not even override manually.

**Pull request closed without merging**